tests/transforms/test_hamiltonian_expand.py

```diff
@@ -259,7 +260,287 @@ def test_hamiltonian_dif_tensorflow(self):
+
+
+with QuantumTape() as s_tape1:
+    qml.PauliX(0)
+    S1 = qml.s_prod(1.5, qml.prod(qml.PauliZ(0), qml.PauliZ(1)))
+    qml.expval(S1)
+    qml.expval(S1)
+    qml.state()
+
+with QuantumTape() as s_tape2:
+    qml.Hadamard(0)
+    qml.Hadamard(1)
+    qml.PauliZ(1)
+    qml.PauliX(2)
+    S2 = qml.op_sum(
+        qml.prod(qml.PauliX(0), qml.PauliZ(2)),
+        qml.s_prod(3, qml.PauliZ(2)),
+        qml.s_prod(-2, qml.PauliX(0)),
+        qml.PauliX(2),
+        qml.prod(qml.PauliZ(0), qml.PauliX(1)),
+    )
+    qml.expval(S2)
+    qml.probs(op=qml.PauliZ(0))
+    qml.expval(S2)
+
+S3 = qml.op_sum(
+    qml.s_prod(1.5, qml.prod(qml.PauliZ(0), qml.PauliZ(1))), qml.s_prod(0.3, qml.PauliX(1))
+)
+
+with QuantumTape() as s_tape3:
+    qml.PauliX(0)
+    qml.expval(S3)
+    qml.probs(wires=[1, 3])
+    qml.expval(qml.PauliX(1))
+    qml.expval(S3)
+    qml.probs(op=qml.PauliY(0))
+
+
+S4 = (
+    qml.prod(qml.PauliX(0), qml.PauliZ(2))
+    + qml.s_prod(3, qml.PauliZ(2))
+    - qml.s_prod(2, qml.PauliX(0))
+    + qml.PauliZ(2)
+    + qml.PauliZ(2)
+    + qml.prod(qml.PauliZ(0), qml.PauliX(1), qml.PauliY(2))
+)
+
+with QuantumTape() as s_tape4:
+    qml.Hadamard(0)
+    qml.Hadamard(1)
+    qml.PauliZ(1)
+    qml.PauliX(2)
+
+    qml.expval(S4)
+    qml.expval(qml.PauliX(2))
+    qml.expval(S4)
+    qml.expval(qml.PauliX(2))
+
+SUM_TAPES = [s_tape1, s_tape2, s_tape3, s_tape4]
+SUM_OUTPUTS = [
+    [
+        -1.5,
+        -1.5,
+        np.array(
+            [
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                1.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+            ]
+        ),
+    ],
+    [-6, np.array([0.5, 0.5]), -6],
+    [-1.5, np.array([1.0, 0.0, 0.0, 0.0]), 0.0, -1.5, np.array([0.5, 0.5])],
+    [-8, 0, -8, 0],
+]
+
+
+class TestSumExpand:
+    """Tests for the sum_expand transform"""
+
+    @pytest.mark.parametrize(("tape", "output"), zip(SUM_TAPES, SUM_OUTPUTS))
+    def test_sums(self, tape, output):
+        """Tests that the sum_expand transform returns the correct value"""
+
+        tapes, fn = sum_expand(tape)
+        results = dev.batch_execute(tapes)
+        expval = fn(results)
+
+        assert all(qml.math.allclose(o, e) for o, e in zip(output, expval))
+
+        qs = QuantumScript(tape.operations, tape.measurements)
+        tapes, fn = sum_expand(qs)
+        results = dev.batch_execute(tapes)
+        expval = fn(results)
+
+        assert all(qml.math.allclose(o, e) for o, e in zip(output, expval))
+
+    @pytest.mark.parametrize(("tape", "output"), zip(SUM_TAPES, SUM_OUTPUTS))
+    def test_sums_no_grouping(self, tape, output):
```

Comment:
Why test this with both tape and qscript?

Reply:
Also I don't think this proves grouping was off? Since we expect the expval to be the same regardless of method right?

Reply:
The tests were copied from `TestHamiltonianExpand`. I believe Christina wanted to test that both `QuantumTape` and `QuantumScript` worked.

But we don't need to do this here, given that this is a new functionality. Removed all the duplicated tests.